src/BizHawk.Emulation.Common/Base Implementations/MemoryCallbackSystem.cs

```diff
@@ -66,47 +66,50 @@ public void Add(IMemoryCallback callback)
 			}
 		}
 
-		private static void Call(MemoryCallbackCollection cbs, uint addr, uint value, uint flags, string scope)
+		private static uint Call(MemoryCallbackCollection cbs, uint addr, uint value, uint flags, string scope)
 		{
+			uint cbReturn = value; //By default, if no callback is called, it will return the value sent by the core unmodified.
 			foreach (var cb in cbs)
 			{
 				if (!cb.Address.HasValue || (cb.Scope == scope && cb.Address == (addr & cb.AddressMask)))
 				{
-					cb.Callback(addr, value, flags);
+					cbReturn = (uint) cb.Callback(addr, value, flags); //If there are many callbacks registered to the same addr, only the last callback's return value will be sent to the core.

```

Comment:
From https://github.com/TASEmulators/BizHawk/pull/3821#discussion_r2023574499, this needs to be something like `if (cb.Callback(addr, value, flags) is uint n) cbReturn = n;` to get the desired behaviour.

Though I wonder how that will behave if one callback "vetos" by returning a value, and another callback uses the old method of poking memory.